test/unit/DolaFlashMinter.test.ts

```diff
@@ -0,0 +1,196 @@
+import { expect } from "chai";
+import { MockContract } from "ethereum-waffle";
+import { BigNumber, constants, Contract, utils } from "ethers";
+import { waffle } from "hardhat";
+import IERC3156FlashBorrowerABI from "../../artifacts/contracts/interfaces/IERC3156FlashBorrower.sol/IERC3156FlashBorrower.json";
+import MockDolaABI from "../../artifacts/contracts/test/MockDola.sol/MockDola.json";
+import DolaFlashMinterABI from "../../artifacts/contracts/test/TestDolaFlashMinter.sol/TestDolaFlashMinter.json";
+import { DolaFlashMinter } from "../../typechain";
+import { ETH } from "../utils/utils";
+
+describe("Dola Flash Minter", function () {
+  let mockDola: Contract;
+  let mockBorrower: MockContract;
+
+  let contract: DolaFlashMinter;
+  let contractAsUser: DolaFlashMinter;
+
+  const provider = waffle.provider;
+  const [deployerWallet, userWallet, treasuryWallet] = provider.getWallets();
+  const { deployMockContract, deployContract } = waffle;
+
+  beforeEach(async () => {
+    mockDola = await deployContract(deployerWallet, MockDolaABI);
+    mockBorrower = await deployMockContract(deployerWallet, IERC3156FlashBorrowerABI.abi);
+    contract = (await deployContract(deployerWallet, DolaFlashMinterABI, [
+      mockDola.address,
+      treasuryWallet.address,
+    ])) as DolaFlashMinter;
+    contractAsUser = contract.connect(userWallet);
+  });
+
+  describe("Flash Loan", async function () {
```

Comment:
Maybe add a test for when the borrower doesn't have enough Dola to pay for fee, or minter contract isn't approved to move borrowers Dola.

Reply:
Done